analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -190,7 +190,13 @@ invocation switch
             _ => context.State.ToArray()
         };
 
-    private static ProgramState[] ProcessEquals(SymbolicContext context, IOperation leftOperation, IOperation rightOperation)
+    private static ProgramState[] ProcessEquals(SymbolicContext context, IOperation leftOperation, IOperation rightOperation) =>
+        context.State.Constraint<BoolConstraint>(leftOperation) is { }
+        && context.State.Constraint<BoolConstraint>(rightOperation) is { }
+            ? ProcessEqualsBool(context, leftOperation, rightOperation)
+            : ProcessEqualsObject(context, leftOperation, rightOperation);
```

Comment:
Let's just inline this.
```suggestion
    private static ProgramState[] ProcessEquals(SymbolicContext context, IOperation leftOperation, IOperation rightOperation) =>
        context.State.Constraint<BoolConstraint>(leftOperation) is { } leftBool
        && context.State.Constraint<BoolConstraint>(rightOperation) is { } rightBool
            ? context.SetOperationConstraint(BoolConstraint.From(leftConstraint == rightConstraint)).ToArray()
            : ProcessEqualsObject(context, leftOperation, rightOperation);
```